compiler/statements.go

```diff
@@ -794,8 +796,14 @@ func (c *funcContext) translateResults(results []ast.Expr) string {
 			values[i] = c.translateImplicitConversion(result, tuple.At(i).Type()).String()
 		}
 		c.delayedOutput = nil
-		return " [" + strings.Join(values, ", ") + "]"
+		rVal = " [" + strings.Join(values, ", ") + "]"
+	}
+	if flatten {
+		resumeCase := c.caseCounter
+		c.caseCounter++
+		c.Printf("/* potential blocking in defer */ $s = %[1]d; case %[1]d:", resumeCase)
```

Comment:
Please keep the pattern that the comment is either empty or has the corresponding non-blocking/non-flattened code.